Create Simple Forms Upload Duplicate and add 21-686c

patrick-brown-oddball opened the pull request:
## Summary

We are adding a form upload to the Accredited Rep Portal. We are borrowing from simple forms but creating a new system for our needs. You can read more about the needs and reasoning here - https://dsva.slack.com/docs/T03FECE8V/F08GCG2AQ1H

We are releasing the app in steps. This is the first. The steps for the backend are -

- *HERE* add 21-686c form to persistent attachments and form upload notification email. We will re-use these already built parts for our own app. We are starting off our application with the 21-686c form as a baseline and then we are going to add the other forms once we are done with the infrastructure
- Make the upload tool ARP compliant - create our own controller with validations for our circumstances such as checking power of attorney for the claimant
- Add new table arp_in_progress_forms so any reps that are also veterans don't affect their own uploads

## Related issue(s)

- vets-website pr - https://github.com/department-of-veterans-affairs/vets-website/pull/35338
- content-build pr - https://github.com/department-of-veterans-affairs/content-build/pull/2485
- collab cycle - https://github.com/department-of-veterans-affairs/va.gov-team/issues/104783

- since we are borrowing from simple forms, the data flow diagram is the same - https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1702671758787/1eee327caf163786a70af458b56128b9e5b3ce2e?sender=954d567f-8d91-47d1-b8a4-ce33e64ea3a2

We will eventually add a power of attorney check on submit to make sure the rep has poa for the veteran they are submitting the form for. We are starting with the 686c form and then expand out to all the forms we need

## Testing done

- [ ] *New code is covered by unit tests*

## What areas of the site does it impact?
- A new page at /representative_form_upload

## Acceptance criteria

- [ ]  I added unit tests and integration tests for each feature

patrick-brown-oddball committed:
arf/representative-form-upload-step-1

amprokop commented on app/models/persistent_attachments/va_form.rb:
```diff
@@ -12,7 +12,8 @@ class PersistentAttachments::VAForm < PersistentAttachment
       '21P-0518-1' => { max_pages: 2, min_pages: 2 },
-      '21P-0516-1' => { max_pages: 2, min_pages: 2 }
+      '21P-0516-1' => { max_pages: 2, min_pages: 2 },
+      '21-686c' => { max_pages: 16, min_pages: 2 }
```
is 16 the right value?

kayline committed:
Add feature flag for DR notification callbacks

patrick-brown-oddball committed:
Merge branch 'master' into arf/representative-form-upload-step-1

patrick-brown-oddball committed:
arf/representative-form-upload-step-1 - fixed syntax error

patrick-brown-oddball committed:
arf/representative-form-upload-step-1 - fixed rubocop locally and found issue

mjknight50 commented:
I'm confused as to what this has to do with 21P-686c?

binq committed:
cookie_id for show_yellow_ribbon_table (#21337)

digitaldrk committed:
Financial Management FormProfile Mocking (#21333)

patrick-brown-oddball committed:
arf/representative-form-upload-step-1 - added completed 686c

pmclaren19 committed:
[CST] [Document Status] - 104847 Added endpoint for getting evidence submissions (#21315)

* Added endpoint evidence submissions, added tests, updated benefits claims controller and tests, updated yml files

* added code owners

* update for linter

* add service tag

* add helper function and update controlelrs to use it, add tests

* added codeowner

* remove method

* undo merge that was pushed

* add refactor to evidence_submissions_controller.rb to use less code

LindseySaari committed:
fix: bump to patched version (#21341)

bosawt committed:
Fixing add_person_proxy bug that doesn't properly search with a family name (#21347)

kayline committed:
New VANotify callback pattern for DR forms (#21214)

* New notification callback pattern

* Use new notification callback behind Flipper

* all info specific to the notification goes in callback_metadata
* no longer using :reference

ryan-mcneil committed:
Add recurring_payment to /payment-history endpoint (#21344)

* Add recurring payment to payment_history

* Update docs

* remove required properties for {} case

patrick-brown-oddball committed:
arf/representative-form-upload-step-1 - added form back into email

patrick-brown-oddball commented:
> I'm confused as to what this has to do with 21P-686c?
So we are adding an upload tool for accredited representatives and we picked that form to start with just as a baseline

rmtolmach committed:
delete unused EVSS ITF VCR cassettes (#21326)

rileyanderson committed:
Fix audit_log enum value (#21354)

RachalCassity committed:
Return empty array when BGS uploaded_document is null (#21355)

emilykim13 committed:
vi-1055 Create UserActionEventsController and query route (#20690)

* vi-1055 Create UserActionEventsController and query route

* Add spec for correct date range

* add serializers and minor changes to useractionevents controller

* timezone fix

* fix typo

* fix typo

* fix serializer to apply to each user_action

* change method for serialization

* add files to codeowners

* rmv extra user_action variables

* add more unit tests to cover date range testing

* add unit tests for user actions pagination

* Filter start_date and end_date params in unit test

* add comment

* reword unit test

* rmv test comments

* rmv test covered by another unit test

* minor changes to unit tests

* add current_page to json response

* rmv comment

* update unit tests to match json response for meta

* add event tpye and identifier, make pagination data private

* pass user_verification object

* move context when not logged in into index describe block

* Standarize subject and call subject

* update factory test to prevent flaky results

* use subject call

* refactor subject call

* remove user actions relationship on user action event serializer

* update tests

mjknight50 committed:
Create 0010-db-schema-changes.md (#21291)

lee-delarm6 committed:
102782 notify veteran cc problem (#21306)

* adding epsappointment worker with va notify

adding epsappointment worker with va notify

* Added module, fixed retry

Added module, fixed retry number of times we're polling for the appointment

* Removing payload, changing call

Removing payload not doing anything with it, changing call to include user

* Change to pass in user, added tests

Change to pass in user, added tests

* Fixed tests

Fixed tests

* Fixed tests

Fixed tests

* updating notify

updating notify

* Fixed tests

Fixed tests

* remove duplicate lines, fix test

remove duplicate lines, fix test

* Set to master branch gemfile.lock

* Update Gemfile.lock

* Update Gemfile.lock

* Fix linting issues

Fix linting issues

* broke test, fixing it

broke test, fixing it

* adding other settings needed

adding other settings needed

* Adding simple custom error

Adding simple custom error

* rubocop fixes

rubocop fixes

* Fixed error in test

Fixed error in test

---------

Co-authored-by: PhilipDeFraties <philip.defraties@gmail.com>

Commit:
PBP 105360 add unit test to check for specific pdf stamps in intake job (#21321)

coope93 committed:
Add model relationships and specs for StdStates and StdInsitutionFacilities models (#21338)

* add specs for std models and add new associations

* add factory for std_institution_facilities

* move specs to correct modules

* rubocop

pixiitech committed:
Migration for icn_temporary_identifiers (#101374) -- #1 (#21231)

* Migration for ar_icn_temporary_identifiers (#101374)

- convert to uuid
- Changes per Oren review

* Add indices

pixiitech committed:
Update ARP dev seed script (#103837) (#21340)

* Update ARP dev seed script (#103837)

* Create PowerOfAttorneyFormSubmissions in seed task

nathanbwright committed:
VANotify - Add callback token (#21342)

khenson-oddball committed:
Iir1516 vre module migration (#21288)

pixiitech committed:
Check for step completion instead of PoA status (#105648) (#21352)

* Check for step completion instead of PoA status (#105648)

* Update success/failure logic

stevenjcumming committed:
add carrierwave upload directories to get ignore (#21358)

dependabot[bot] committed:
Bump datadog from 2.10.0 to 2.12.2 (#21334)

Bumps [datadog](https://github.com/DataDog/dd-trace-rb) from 2.10.0 to 2.12.2.
- [Release notes](https://github.com/DataDog/dd-trace-rb/releases)
- [Changelog](https://github.com/DataDog/dd-trace-rb/blob/master/CHANGELOG.md)
- [Commits](https://github.com/DataDog/dd-trace-rb/compare/v2.10.0...v2.12.2)

---
updated-dependencies:
- dependency-name: datadog
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

dependabot[bot] committed:
Bump parallel_tests from 4.9.0 to 5.1.0 (#21336)

Bumps [parallel_tests](https://github.com/grosser/parallel_tests) from 4.9.0 to 5.1.0.
- [Changelog](https://github.com/grosser/parallel_tests/blob/master/CHANGELOG.md)
- [Commits](https://github.com/grosser/parallel_tests/compare/v4.9.0...v5.1.0)

---
updated-dependencies:
- dependency-name: parallel_tests
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

dependabot[bot] committed:
Bump pundit from 2.4.0 to 2.5.0 (#21335)

Bumps [pundit](https://github.com/varvet/pundit) from 2.4.0 to 2.5.0.
- [Changelog](https://github.com/varvet/pundit/blob/main/CHANGELOG.md)
- [Commits](https://github.com/varvet/pundit/compare/v2.4.0...v2.5.0)

---
updated-dependencies:
- dependency-name: pundit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

jerekshoe committed:
[Letters] Fix issue with `benefit_summary_dependent` letter (#21353)

* Removing the veteran_sponsor_resolver as it is no longer needed

* Removing spec file

* Removing veteran_sponsor_resolver from mobile controller

* Removing veteran_sponsor_resolver import

* Updating CODEOWNERS

* Removing unneeded spec examples

* Fixing test

rmtolmach committed:
Change PPIU wording to LH form 526 (#21360)

kayline committed:
Add feature flag for form 4142 schema validation (#21330)

coope93 committed:
[10-10EZ] Add HealthFacililitiesImportJob (#21258)

* add factory for std_institution_facilities

* rubocop

* wip for adding new health_facilities_import_job

* Simple Forms Notification Email refactor (#21021)

* use new notification files

* Pull template_ids out to a yml file

* tests

* test

* Alphabetize template ids

* chore: Remove unused EVSS::ReferenceData::IntakeSitesResponse class (#21189)

* bump ruby-saml to 1.18.0 (#21229)

* BE | Ask VA Api: update crm production env (#21223)

* update crm production env

* Add new keys to setting

* fix settings order

---------

Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

* Add breakers integration to UnifiedHealthData service (#21178)

* Add breakers integration to UnifiedHealthData service

* Add breakers middleware to Faraday connection in UnifiedHealthData configuration

* Burial 104893 reorganize module lib (#21192)

* update veft-prod to veft (#21245)

* 4487 - Added ST schema and spec files (#21207)

* Added ST schema and spec files

* Changed spec name and fixed lint errors

* Removed before block

* trigger health_facilities_import_job in std_institution_import_job behind toggle

* update codeowners for health_facility

* update codeowners for new factories

* call get_facilities in import job so names are normalized

* remove get_facility_ids from v2 client

* add retries and statsd increments to health import job

* add HealthFacility model

* Revert "add HealthFacility model"

This reverts commit a54250672af400201a461394a4d7504411515f4c.

* Revert "Revert "add HealthFacility model""

This reverts commit 9c9c897eec3f05e4a05e2a3c01bd71576a15a9f4.

* removed extra line on end of file from other team

* add back health_facility facotry

* remove std_instituion_facilities factory since it is in a module

* use association in import job query

---------

Co-authored-by: Eric Tillberg <Thrillberg@users.noreply.github.com>
Co-authored-by: Eric Boehs <eric@boehs.com>
Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>
Co-authored-by: Khoa Nguyen <134089461+Khoa-V-Nguyen@users.noreply.github.com>
Co-authored-by: Adrian Rollett <101649+acrollet@users.noreply.github.com>
Co-authored-by: Wayne Weibel <wayne.weibel@adhocteam.us>
Co-authored-by: brendan-git-va <brendan.dunn@va.gov>

dependabot[bot] committed:
Bump dry-schema from 1.14.0 to 1.14.1 (#21367)

Bumps [dry-schema](https://github.com/dry-rb/dry-schema) from 1.14.0 to 1.14.1.
- [Release notes](https://github.com/dry-rb/dry-schema/releases)
- [Changelog](https://github.com/dry-rb/dry-schema/blob/main/CHANGELOG.md)
- [Commits](https://github.com/dry-rb/dry-schema/compare/v1.14.0...v1.14.1)

---
updated-dependencies:
- dependency-name: dry-schema
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

dependabot[bot] committed:
Bump blind_index from 2.6.1 to 2.6.2 (#21366)

Bumps [blind_index](https://github.com/ankane/blind_index) from 2.6.1 to 2.6.2.
- [Changelog](https://github.com/ankane/blind_index/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ankane/blind_index/compare/v2.6.1...v2.6.2)

---
updated-dependencies:
- dependency-name: blind_index
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

michaelclement committed:
104848 - Update IvcChampvaForm model with new fields for VES integration - IVC CHAMPVA forms (#21293)

* added migration for ivcchampvaform to add new fields

* removed test

* regenerated schema.db to get appropriate header

michaelclement committed:
104641a - Remove `champva_confirmation_email_bugfix` flipper - IVC CHAMPVA forms (#21319)

* removed champva_confirmation_email_bugfix flipper

* linting

dependabot[bot] committed:
Bump rubocop-factory_bot from 2.26.1 to 2.27.1 (#21365)

Bumps [rubocop-factory_bot](https://github.com/rubocop/rubocop-factory_bot) from 2.26.1 to 2.27.1.
- [Release notes](https://github.com/rubocop/rubocop-factory_bot/releases)
- [Changelog](https://github.com/rubocop/rubocop-factory_bot/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rubocop/rubocop-factory_bot/compare/v2.26.1...v2.27.1)

---
updated-dependencies:
- dependency-name: rubocop-factory_bot
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

jefftmarks committed:
EDM/maintenance window lighthouse education benefits (#21369)

* add service

* alphebatize

* dev and test

kjsuarez committed:
103011 add 'assistive devices' to MDOT eligibility (#21312)

* 103011 add 'assistive devices' to MDOT eligibility

* 103011 end of file new line

Commit:
Burial 105362 remove old model code (#21310)

robertroddev committed:
[mhv-68364] fix tooltip bug so counter increments per unique session, also allow counter to be directly updated via request body. (#21359)

Co-authored-by: stevenjcumming <134282106+stevenjcumming@users.noreply.github.com>

dfong-adh committed:
104037: Add feature flags for DR notification callbacks (#21332)

ryan-mcneil committed:
Fix payment_history API docs (#21373)

* Fix date response format in API docs

* generate_docs

pixiitech reviewed (dismissed):
looks good

stevenjcumming commented:
@patrick-brown-oddball I think you may have pulled some other PRs into your branch. You may want to close this PR and open a new one
